component/authenticator/src/main/java/org/wso2/carbon/identity/application/authenticator/totp/util/TOTPAuthenticatorCredentials.java

```diff
@@ -391,20 +390,15 @@ public boolean authorizeAndStoreSecret(int verificationCode, String username) {
 		}
 	}
 
-	private void storeSecretKey(String secretKey, String username) {
+	private void storeSecretKey(String secretKey, String username, UserRealm userRealm) {
 
 		Map<String, String> userClaims = new HashMap<>();
 		try {
-			UserRealm userRealm = TOTPUtil.getUserRealm(username);
-			if (userRealm != null) {
-				userClaims.put(TOTPAuthenticatorConstants.SECRET_KEY_CLAIM_URL, TOTPUtil.encrypt(secretKey));
-				userRealm.getUserStoreManager().setUserClaimValues(username, userClaims, null);
-			}
+			userClaims.put(TOTPAuthenticatorConstants.SECRET_KEY_CLAIM_URL, TOTPUtil.encrypt(secretKey));
+			userRealm.getUserStoreManager().setUserClaimValues(username, userClaims, null);
```

Comment:
Don't we need to pass tenant aware user name here?

Reply:
in line 371 we are passing the tenant aware name when updating the user claims

Reply:
we are passing the tenant aware username [line 377](https://github.com/wso2-extensions/identity-outbound-auth-totp/pull/150/files#diff-97f408b1866a1e96ceada8461e6c432a5b71b234abcb9ae16db064156153cab8R377). Will change the variable name here to avoid the confusion